### PR TITLE
L4: data structures

### PR DESCRIPTION
**Pull request by Brandonz07**

### CODE REVIEW

File: src/br/js/L4-data-structures/The_sum_of_a_range.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  } else {
+    for (let i = start; i >= end; i += step) {
+      arr.push(i);
+    }
+  }
+  return arr;
+}
+
+function sum(arr) {
+  let sum = 0;
```
Comment (carin91) [inline src/br/js/L4-data-structures/The_sum_of_a_range.js:16]: refactor using reduce method for arrays.

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/br/js/L4-data-structures/The_sum_of_a_range.js (hunk 1 above)
Comment (Brandonz07) [reply src/br/js/L4-data-structures/The_sum_of_a_range.js]: i use reduce

File: src/br/js/L4-data-structures/A_list.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  let list = null;
+  for (let i = array.length - 1; i >= 0; i--) {
+    list = { valor: array[i], resto: list };
+  }
+  return list;
+}
+
+function listToArray(list) {
+  let array = [];
+  for (let node = list; node; node = node.resto) {
```
[the remark this answers is not in the document]
Comment (Brandonz07) [reply src/br/js/L4-data-structures/A_list.js]: avoid use

File: src/br/js/L4-data-structures/A_list.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    array.push(node.valor);
+  }
+  return array;
+}
+
+function prepend(valor, list) {
+  return { valor, resto: list };
+}
+
+function nth(list, n) {
```
[the remark this answers is not in the document]
Comment (Brandonz07) [reply src/br/js/L4-data-structures/A_list.js:22]: avoid use

File: src/br/js/L4-data-structures/Deep_comparison.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    if (keyA.length != keyB.length) return false;
+
+    for (let key of keyA) {
+      if (!keyB.includes(key) || !deepEqual(a[key], b[key])) return false;
+    }
+
+    return true;
+  } else {
+    return a === b;
```
[the remark this answers is not in the document]
Comment (Brandonz07) [reply src/br/js/L4-data-structures/Deep_comparison.js:16]: to compare simple values

**Review by carin91 — APPROVED** [review]

**Pull request merged**